### PR TITLE
Completed Step4

### PR DESCRIPTION
**Pull request by yukinkm0**
## What
json fileからSQLiteにデータを移行、商品の検索機能を追加、カテゴリの情報を別のテーブルに移しました

## CHECKS :warning:

Please make sure you are aware of the following.

- [x] **The changes in this PR doesn't have private information

### CODE REVIEW

File: db/items.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE TABLE items(id INTEGER PRIMARY KEY, name TEXT, category TEXT, image_name TEXT);
```
Comment (mshibuya) [inline db/items.sql]: You forgot to update the `category` column, right?
Also please include the `CREATE TABLE` statement for the `categories` table.

File: python/main.py [hunk 2]
```diff
@@ -21,7 +22,8 @@
     allow_headers=["*"],
 )
 
-json_file="./items.json"
+json_file = "./items.json"
```
Comment (mshibuya) [inline python/main.py]: This is no longer used.

File: python/main.py [hunk 3]
```diff
@@ -48,25 +50,76 @@ def root():
...
 
 
 @app.get("/items")
 def get_items():
-    with open(json_file, mode='r') as getfile:
-        items = json.load(getfile)
-    return items
+    # with open(json_file, mode='r') as getfile:
+    #     items = json.load(getfile)
+    # return items
```
Comment (mshibuya) [inline python/main.py]: Please don't keep the commented-out code, as it decreases readability.
We're keeping history using Git and you can retrieve old code if it becomes necessary.

File: python/main.py [hunk 4]
```diff
@@ -86,11 +139,36 @@ async def get_image(image_name):
...
+@app.get("/search")
+def search_item(keyword: str):
+    logger.info(f"Search item name: {keyword}")  # keyword探す
+
+    # connect to db
+    conn = sqlite3.connect(db)
+    cur = conn.cursor()
+
+    # 検索
+    cur.execute('SELECT * FROM items WHERE name LIKE ?', (keyword,))
```
Comment (mshibuya) [inline python/main.py]: Does this really performs the partial match, as suggested in the instruction?

> 指定したキーワードを含む商品一覧を返す

File: python/main.py [hunk 5]
```diff
@@ -86,11 +139,36 @@ async def get_image(image_name):
...
+    # with open(json_file, mode='r') as j:
+    #     items = json.load(j)
 
     if item_id >= len(items['items']):
         raise HTTPException(status_code=404, detail="No item found with this id")
     
     else: 
+        # close db
+        conn.close()
         return items['items'][item_id - 1]
```
Comment (mshibuya) [inline python/main.py]: Please retrieve this from the DB as well!

**Review by mshibuya — COMMENTED** [review]
You're almost there! 👍

File: db/items.sql (hunk 1 above)
Comment (yukinkm0) [reply db/items.sql]: I've added the table "category", and edited the table "items"

File: python/main.py (hunk 2 above)
Comment (yukinkm0) [reply python/main.py]: Erased that line.

File: python/main.py (hunk 3 above)
Comment (yukinkm0) [reply python/main.py]: Ok. I erased all of the commented-out code.

File: python/main.py (hunk 4 above)
Comment (yukinkm0) [reply python/main.py]: I rewrote ```(keyword)``` to ```(f'%{keyword}%',)```.

File: python/main.py (hunk 5 above)
Comment (yukinkm0) [reply python/main.py]: rewrote
```   conn = sqlite3.connect(db)
    cur = conn.cursor()

    cur.execute('SELECT * FROM items WHERE category_id ?', (item_id,))
    get_item_id = cur.fetchone()

    # close cur
    cur.close()
    # close db
    conn.close()

    #itemが存在しない場合
    if not get_item_id:
        raise HTTPException(status_code=404, detail="No item found with this id")
    
    return get_item_id

**Pull request merged**